### PR TITLE
VC-4221 processing add affiliate

### PR DESCRIPTION
**Pull request by alex-bodnar**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/accounts/add_affiliate.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	if _, err := r.db.NamedExec(query, toAddAffiliatesDatabase(msg...)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetAddAffiliate - get the given affiliate from database
+func (r Repository) GetAddAffiliate(accfilter filter.Filter) ([]*accountstypes.MsgAddAffiliate, error) {
```
Comment (IamStubborN) [inline database/vipcoin/chain/accounts/add_affiliate.go]:
```suggestion
func (r Repository) GetAddAffiliate(accountFilter filter.Filter) ([]*accountstypes.MsgAddAffiliate, error) {
```

File: _tests/database/accounts/add_affiliate_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	db, err := sqlx.Connect("pgx", "host=localhost port=5432 user=postgres dbname=juno password=postgres sslmode=disable")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the codec
+	codec := simapp.MakeTestEncodingConfig()
+
+	type args struct {
+		accfilter filter.Filter
```
Comment (IamStubborN) [inline _tests/database/accounts/add_affiliate_test.go]:
```suggestion
		accountFilter filter.Filter
```

**Review by IamStubborN — APPROVED** [review]

File: modules/vipcoin/chain/accounts/msg_server_add_affiliate.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	case err != nil:
+		return err
+	case len(accountArr) != 1:
+		return types.ErrInvalidHashField
+	}
+
+	account := accountArr[0]
+	affiliate := affiliateArr[0]
+
+	accIsSet := false // The account affiliates record is already set
```
Comment (DirusK) [inline modules/vipcoin/chain/accounts/msg_server_add_affiliate.go]: var accountIsSet bool

File: modules/vipcoin/chain/accounts/msg_server_add_affiliate.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	for _, a := range account.Affiliates {
+		if a.Address == affiliate.Address {
+			accIsSet = true
+		}
+	}
+
+	affIsSet := false // The affiliate account affiliates record is already set
```
Comment (DirusK) [inline modules/vipcoin/chain/accounts/msg_server_add_affiliate.go]: var accountIsSet bool

File: modules/vipcoin/chain/accounts/msg_server_add_affiliate.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	for _, a := range affiliate.Affiliates {
+		if a.Address == account.Address {
+			affIsSet = true
+		}
+	}
+
+	if !accIsSet {
+		a := &types.Affiliate{
```
Comment (DirusK) [inline modules/vipcoin/chain/accounts/msg_server_add_affiliate.go]: affiliate, not 'a'

**Comment by DirusK** [comment]
LGTM

**Pull request merged**